### ISSUE TITLE
tagsrvcs.com

### ISSUE THREAD

**Issue by massar** [EFForg/privacybadgerchrome#431]
http://s.tagsrvcs.com/2/4.9.4/loaded.js
http://s.tagsrvcs.com/2/825542/analytics.js

are loaded from:
 http://assets.adobedtm.com/6372cf21ef88ee60bc2977a4898dcb5c7945a212/scripts/satellite-5579cf7562663803ed570200.js
http://assets.adobedtm.com/6372cf21ef88ee60bc2977a4898dcb5c7945a212/scripts/satellite-554bd4186631632069aa0300.js

Which then calls up on http://s.tagsrvcs.com/<lots of params> to actually log stuff.

they log every 5 seconds what the heck you are doing on arstechnica.com
Seems arstechnica just needs to become a no-js site with the various trackers they are loading....

**Comment by cooperq** [comment]
@massar pardon me, but I'm not sure I understand what the bug you are reporting here is. Could you rephrase, perhaps?

**Comment by massar** [comment]
Go to any arstechnica.com article, check the 'network' tab of Chrome and see that the above tracking stuff is being loaded.

**Comment by SwartzCr** [comment]
in future versions of privacy badger you will have the ability to block third parties that aren't serving cookies. If you want to use the experimental version of privacy badger you can pull from the git repo now, otherwise we should have a new release out in a few weeks!

**Comment by massar** [comment]
It would be good if there is a "Limitations" section in the readme that describes what PB does and what it does not block. Which could then also state 'X blocked starting at version N".

**Comment by SwartzCr** [comment]
The issue privacy badger is trying to solve is data aggregation across multiple first party sites. It's not as important to use to block tracking across a single first party domain because that first party domain also clearly can track you. If Ars technica allows a third party that sees what you're doing on ars technica, this seems no different than ars collecting their own analytics, and presumably it is ars's choice. The issue comes when that same thrid party knows what you're doing on other first party domains, and that is what Privacy Badger blocks. This should all be explained in our (upcoming) expanded onboarding site

**Comment by massar** [comment]
> It's not as important to use to block tracking across a single first party domain because that first party domain also clearly can track you.

The site itself is always the one that decides to track you. Thus under that comment, PB will not disable tracking for anything.

For instance, everyone knows that a little Twitter "Tweet count" or a Facebook "Like" or a Google "Plus" icon on the site also causes tracking (due to referrer and cookies, and well a connection).

But would for instance:
8<--------
https://twitter.com/incloud/status/619624021123010560

"WebRTC being used now by embedded 3rd party on http://nytimes.com   to report visitors' local IP addresses. "
-------->8

NYTimes clearly decided to add that to their site, thus enabling tracking of their users.

Be something that Privacy Badger will block? If not, then what is the point of PB?

**Comment by michael-oneill** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

As well as your local IP address being reported there is also a (browser created) persistent UID used by Media Capture & Streams (deviceId), which is also accessible third-party JavaScript in a drive-by attack.

https://lists.w3.org/Archives/Public/public-privacy/2015AprJun/0080.html



Also if a tracker controls the DNS Nameserver for a domain it can create a wildcard entry for a STUN server and track that way. See:

https://www.w3.org/wiki/Privacy/IPAddresses#Using_wildcard_DNS_entries_as_persistent_identifiers


Mike

-----BEGIN PGP SIGNATURE-----
Version: GnuPG v1
Comment: Using gpg4o v3.4.103.5490 - http://www.gpg4o.com/
Charset: utf-8

iQEcBAEBAgAGBQJVoMS8AAoJEHMxUy4uXm2JxqoH/2m8UF1hNe2vRpE1jR1Tm32j
OMO8887EYXk0ro/K8JXC/FrlWgtxAtJVo9yhlCzBJb4Briwu4oYSt86nY5P7X21Z
KRjOM277kOB74eG3fcpzfJEkOhknMOF/K6COheSWQOq4YyahmJJC7UmAt2TixMLq
7DCLJme7eoDkbUPFSBbIfLssZLTvffr1lw0CzyzvtJrxyH5Z2l1wc3aOzUjXQOk8
XKyuGwwBGpwmIRocwoiZ5w9p4lhk7HnkPo5jGyEtsF50jILjZBHALH1Hfp1mo/Ae
eLi+AOej+70eq8Bt71NLD9BPKG0KUFZ1jNYN9ANC5ETN7lP+OBWgaoiDL+o77fA=
=rswT
-----END PGP SIGNATURE-----

**Comment by dakami** [comment]
This is part of an anti-bot technology I've been developing at White Ops (whiteops.com) for some time.  There's a flip side to privacy here; turns out something like 2/3rds of bot fraud comes from home users who get compromised so as to effect more ad fraud.  We're basically attacking the funding channel that gets people hacked.  But it does require us to be able to detect the hacking, so we have these tests deployed.

We don't have any persistent IDs in our system; they don't actually help with bot detection because bots have all the IDs (which has been a problem for years).

**Comment by jwilkins** [comment]
@dakami Which part is whiteops? The WebRTC, the extra 12 requests/minute or both?

**Comment by dakami** [comment]
Both.  We're getting some interesting things where we get embedded a pile
of times on a page and we're adapting the code to deal with that.

**Comment by jwilkins** [comment]
@dakami you get how that's a dick move, right?  Compromising user privacy and security in order to make ad view attribution better for companies.

**Comment by dakami** [comment]
Two thirds of ad fraud is coming from home users.  All those ways home
users get hacked -- drive bys, app wrappers, codec packs, fake avs -- guess
what's paying for all these botnets.

We don't do anything to legit users -- not even a CAPTCHA.  But we stop
checks from being written to people getting code execution on their boxes.
I think that's pretty cool.

**Comment by jwilkins** [comment]
No doubt.  But let me understand your argument, you are saying that you're going to wipe out botnets with invasive user enumeration and you're going to do it with features of the browser and nothing underhanded?

You don't think that they'll just improve their tools so that you can't detect them again?  You know, the way that the cat and mouse game has gone the last couple of decades.

**Comment by dakami** [comment]
No absolutes in this game, you know that painfully well.  But we can (and
are) certainly biasing ourselves towards busting bots and away from
violating user privacy.  The bots actually have all the user PII, so it
doesn't inform us at all if it's present or absent.  It's one of many
reasons we don't even use persistent storage/cookies.

**Comment by jwilkins** [comment]
Away from violating user privacy?  You're polling 12x a minute and leaking the user's IP.  It doesn't matter that you're not also hoovering their assorted cookies and so forth, you are assisting the  services that *do* create them.  If your product works, you are making increasing the accuracy of the tracking and making it cheaper to collect this data. Not just for advertisers, but for the intelligence community which we know piggybacks off this type of thing.

If you presume that the box is compromised and a rootkit is more than possible, you're in the position of trying to identify malicious behavior from a constrained sandbox in the browser.  You are trading off user privacy for a battle that you will lose as soon as your product gets enough traction to affect the botnet owners.

Dan, I hope that this is just a case of tunnel vision and that you'll step back and take a look at what you're building.

**Comment by jacobsoo** [comment]
Seems like washingtonpost.com, nbcnews.com, ft.com , cnbc.com & bloomberg.com all uses the same 3rd party tracking code from s.tagsrvcs.com too. O.o"

**Comment by jacobsoo** [comment]
Seems like mostly American website(s) are using that.
http://www.vanityfair.com/ and http://www.wired.com/ are using the same3rd party tracking code.

Apparently Eric Lawrence found an interesting thing if the content is blocked in IE.
http://textslashplain.com/2015/06/22/content-blocking-unintended-consequences/

**Comment by michael-oneill** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

Hi Jacob,

That was quick.

As of about an hour ago there are no longer STUN accesses from nytimes.com, washingtonpst.com, ft.com and cnbc.com. In case it is a geolocation specific thing what are you seeing? I am in UK

Mike

**Comment by dakami** [comment]
Indeed, I killed the STUN queries.

--Dan

**Comment by jwilkins** [comment]
@dakami Thank you! If you ever want a fresh set of eyes on this sort of thing, ping me, we can debate it over coffee/drinks.

**Comment by jacobsoo** [comment]
Indeed that previous WebRTC script is removed. Now they are using another script for tracking though.
Haven't look whether PrivacyBadger will work well against that though.

The script is seen at http://bloomberg.com
http://pastebin.com/KV1YBDrp

**Comment by dakami** [comment]
"They" is me!  Happy to jump on a call and discuss.

**Comment by jacobsoo** [comment]
@dakami Hi sir, just a quick question here.
But i took a quick glimpse at the Javascript, did you check whether certain files exist?
Like how those anti-Emulator techniques? Just wondering is that enough for detecting bots without intruding into privacy since i only saw VirtualBox and VMWare?

**Comment by michael-oneill** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA1

Thanks for your fast response Dan. The previous code has illuminated important privacy risks with this otherwise useful API.

Mike

**Comment by Ammond** [comment]
(Dakami) Dan,

The functionality used by s.tagsrvcs.com causes transparent authentication to fail. This affects all entities, especially enterprises that use transparent authentication. The simple fix is to set s.tagsrvcs.com to NOT authenticate or use explicit auth.

Can the code be written to work well with transparent auth? In that mode redirection is used to get auth. Browsers pop-up a auth panel to enter credentials for s.tagsrvcs.com. Users think their workstations have been compromised.
